lib/action_controller/serialization.rb

```diff
@@ -56,10 +56,15 @@ def use_adapter?
 
     [:_render_option_json, :_render_with_renderer_json].each do |renderer_method|
       define_method renderer_method do |resource, options|
-        options.fetch(:serialization_context) do
-          options[:serialization_context] = ActiveModelSerializers::SerializationContext.new(request, options)
+        options.freeze
+        serialization_options = options.deep_dup
```

Comment:
@bf4 I'm not sure if it is good to `deep_dup` here as it also `dup` the serializer when present and that made the tests fail. That's why I had to do the workaround below. Is this really needed is your opinion?